### DISCUSSION TITLE
Buildprobleme mit platform = espressif32@6.3.0

### DISCUSSION THREAD

**Discussion by PJuergen** [tbnobody/OpenDTU#950] [Q&A] [answered]
Hallo,
ich habe ein kleines Problem beim Build mit VSCode und der letzten Version vom 22.05.2023.
Und zwar erhalte ich nur Fehlermeldungen beim Build. Grund scheint die Version 6.3.0 von espressif32 zu sein. Ändere ich es wieder zurück in die alte Version espressif32@6.1.0 ist alles in Ordnung.
Kennt jemand das Problem und hat dazu eine Lösung?

Jürgen

Hier mal den Anfang der Fehlermeldungen:

Compiling .pio\build\generic\src\InverterSettings.cpp.o
In file included from C:/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/freertos/port/xtensa/include/freertos/FreeRTOSConfig_arch.h:40,
                 from C:/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/freertos/include/esp_additions/freertos/FreeRTOSConfig.h:16,
                 from C:/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/freertos/include/freertos/FreeRTOS.h:57,
                 from C:/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:33,
                 from include/Configuration.h:4,
                 from src/Configuration.cpp:5:
C:/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/freertos/port/xtensa/include/freertos/xtensa_config.h:40:10: fatal error: xtensa/hal.h: No such file or directory
 #include <xtensa/hal.h>
          ^~~~~~~~~~~~~~
compilation terminated.
Compiling .pio\build\generic\src\Led_Single.cpp.o
Compiling .pio\build\generic\src\MessageOutput.cpp.o
Compiling .pio\build\generic\src\MqttHandleDtu.cpp.o
Compiling .pio\build\generic\src\MqttHandleHass.cpp.o
*** [.pio\build\generic\src\Configuration.cpp.o] Error 1
In file included from C:/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/freertos/port/xtensa/include/freertos/FreeRTOSConfig_arch.h:40,
                 from C:/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/freertos/include/esp_additions/freertos/FreeRTOSConfig.h:16,
                 from C:/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/freertos/include/freertos/FreeRTOS.h:57,
                 from C:/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:33,
                 from include/Configuration.h:4,
                 from src/Datastore.cpp:6:
C:/.platformio/packages/framework-arduinoespressif32/tools/sdk/esp32/include/freertos/port/xtensa/include/freertos/xtensa_config.h:40:10: fatal error: xtensa/hal.h: No such file or directory
 #include <xtensa/hal.h>
          ^~~~~~~~~~~~~~
compilation terminated.

**Comment by tbnobody** [comment]
Da hast du vmtl. nicht gewartet bis der update job von vscode das Update heruntergeladen und extrahiert hat. 

Würde zuerstmal ein Clean mit anschließendem Re-Compile versuchen. Wenn das nicht klappt den .pio ordner löschen. (Wird automatisch neu erstellt). Wenn das auch nicht klappt den 6.3er Ordner unter `C:/.platformio/packages/` löschen.

**Reply by PJuergen** [reply]
Vielen Dank, hat leider nicht geklappt. Gleiches Ergebnis.

**Comment by PJuergen** [comment, marked as answer]
Hallo, Erledigt.
Alles aus dem Ordner "C:\.platformio\packages" gelöscht.
Danach wurde alles neu installiert und jetzt klappt's
